File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        size_t minLength = strs[0].size();
        for (const string& s : strs) {
            if (s.size() < minLength) {
                minLength = s.size();
            }
        }

        string retString = "";

        for (int j = 0; j < minLength; ++j) {
            char curChar = strs[0][j];

            for (int i = 1; i < strs.size(); ++i) {
                if (strs[i][j] != curChar) {
                    return retString;
                }
            }

            retString += curChar;
        }

        return retString;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        minLength = len(strs[0])
        for s in strs:
            if len(s) < minLength:
                minLength = len(s)
        
        retString = ""

        for j in range(minLength):
            curChar = strs[0][j]

            for i in range(1, len(strs)):
                if strs[i][j] != curChar:
                    return retString
            
            retString += curChar
        
        return retString